elasticsearch/rest/src/main/java/org/apache/metamodel/elasticsearch/rest/ElasticSearchRestUpdateCallback.java

```diff
@@ -135,20 +140,32 @@ public void execute(final ActionRequest action) {
 
     private void executeBlocking(final ActionRequest action) {
         try {
-            if (action instanceof PutMappingRequest) {
-                getDataContext().getElasticSearchClient().createMapping((PutMappingRequest) action);
+            final RestHighLevelClient client = getDataContext().getRestHighLevelClient();
+            final ActionResponse result;
+
+            if (action instanceof BulkRequest) {
+                result = client.bulk((BulkRequest) action, RequestOptions.DEFAULT);
+            } else if (action instanceof IndexRequest) {
+                result = client.index((IndexRequest) action, RequestOptions.DEFAULT);
+            } else if (action instanceof DeleteRequest) {
+                result = client.delete((DeleteRequest) action, RequestOptions.DEFAULT);
+            } else if (action instanceof ClearScrollRequest) {
+                result = client.clearScroll((ClearScrollRequest) action, RequestOptions.DEFAULT);
+            } else if (action instanceof SearchScrollRequest) {
+                result = client.scroll((SearchScrollRequest) action, RequestOptions.DEFAULT);
```

Comment:
That's... Rather nasty. It seems to do nothing, why is this needed?

Reply:
Ah, I'm blind, it's of course not the same method called. Still looks terrible, though. :)

Reply:
(I guess an adapter with overloaded methods would be more readable, but it would also be more code)